packages/react/src/components/DataTable/Table.js

```diff
@@ -30,12 +31,20 @@ export const Table = ({
     [`${prefix}--data-table--zebra`]: useZebraStyles,
     [`${prefix}--data-table--static`]: useStaticWidth,
     [`${prefix}--data-table--no-border`]: !shouldShowBorder,
+    [`${prefix}--data-table--sticky-header`]: stickyHeader,
   });
-  return (
+  const table = (
     <table {...other} className={componentClass}>
       {children}
     </table>
   );
+  return stickyHeader ? (
+    <section className={`${prefix}--data-table_inner-container`}>
```

Comment:
Our `<Table>`, `<TableHead>`, etc. tends to be simple components that corresponds to native `<table>`/`<thead>`, etc., respectively. That said, would you add such enhancements to `<TableContainer>`?

Reply:
I agree - however, there is one problem with adding the enhancement to `<TableContainer>`:
While `<Table>` gets the `DataTable`'s `stickyHeader` props thru `getTableProps()`, `<TableContainer>` does not have access to the `stickyHeader` property. That said I see some options:
1. Expect the developer to add the `stickyHeader` both props to `<DataTable>` and `<TableContainer>`. This is not really a good option since if we miss one the table is not as expected.
2. Always insert 
```
<section className={`${prefix}--data-table_inner-container`}>
      {children}
    </section>
```
below `TableContainer`. This sounds like a good solution, but it would modify all existing tables adding the `section` with the `${prefix}--data-table_inner-container` style:
```
  .#{$prefix}--data-table_inner-container {
    background-color: $ui-03;
    padding-top: rem(48px);
    transform: translateZ(0);
  }
```
The most obvious change is the top padding. I assume backward compatibility is a sacred thing, so that does not look like a good option.
3. Use `getTableProps()` in `TableContainer` to get the `stickyHeader` prop. Feels like we are using `getTableProps()` in a non-expected way though.
4. Exclusively pass the `stickyHeader` prop down from `DataTable` to all children. That would allow `TableContainer` to use it.

My feeling is that option 4 is the best option. Will go down that way for now.

Reply:
Thank you for pointing this out @gmoehler! This is happening because `<DataTable>` when its originally created didn't anticipate any props/state to be propagated to `<TableContainer>`. To cope with what you found, I have two suggestions:

1. Instead of adding `stickyHeader` prop to `<DataTable>`/`<Table>`, we simply add `stickyHeader ` to `<TableContainer>` and let application to specify `stickyHeader ` to `<TableContainer>`
2. Introduce `getTableContainerProps` in a similar manner as [`getTableProps`](https://github.com/carbon-design-system/carbon/blob/v10.5.0/packages/react/src/components/DataTable/DataTable.js#L567)

I'd be inclined to 1. because `<DataTable>` is merely a state manager (instead of one working as the table UI).

Reply:
@asudoh, thanks for your suggestion. Even if we add the `stickyHeader` prop to `<TableContainer>`, we still need the `stickyHeader` property on table (to add the css class). 
Would you suggest to add it to `getTableProps()` additionally, or should we pass it down from `<TableContainer>` to all its children. I would prefer to only specify it once, which would imply the second solution.

Reply:
For the suggested solution the code in `TableContainer` could look something like this:
```
    const childrenWithStickHeader = React.Children.map(this.props.children, child => 
            React.cloneElement(child, {stickyHeader: this.props.stickyHeader});
    );
    return (
        <div {...rest} className={tableContainerClasses}>
            {title && (...)}
            {childrenWithStickHeader}
        </div>

```

Reply:
I see, my apologies for not being clear enough. My 2nd suggestion was adding `getTableContainerProps()` or a like.

Reply:
ok... this looks good to me. 
And if i understand the concept correctly we can pass the `this.props.stickyHeader` from DataTable down to the `TableContainer` via `getTableProps()` and to `Table` via `getTableProps()`. Agreed?

Reply:
I worked on the implementation using `getTableContainerProps()` only to find out that the assumption that the inner container
```
<section className={`${prefix}--data-table_inner-container`}>...</section>
```
needs to go below `TableContainer` is just true for the simplest case (e.g. without pagination and alike). 

In fact what the complex example https://www.carbondesignsystem.com/components/data-table/code#data-table-with-pagination shows is that it in fact needs to wrap `<Table>`, whereas Toolbar Content needs to be outside of the inner container.
So, are we back to my initial solution? 
Maybe we should check with @jnm2377 to see what exactly would be the correct rule to put the inner container.